### PR TITLE
feat: billing shipping addresses checkout

### PR DESCRIPTION
**Pull request by FabianGerke**

### CODE REVIEW

File: apps/web/components/CheckoutAddress/CheckoutAddress.vue [hunk 1]
```diff
@@ -37,17 +45,55 @@
...
+const { saveShippingAddress } = useShippingAddress();
+const { data: activeShippingCountries, getActiveShippingCountries } = useActiveShippingCountries();
+await getActiveShippingCountries();
+
+const props = defineProps<CheckoutAddressProps>();
+const editMode = ref(false);
+const selectedAddress = computed(() => props.addresses?.[0]?.id?.toString() || '');
+const emit = defineEmits(['on-saved']);
+
+function create() {
```
Comment (ksted) [inline apps/web/components/CheckoutAddress/CheckoutAddress.vue:79]:
```suggestion
const create = () => {
```

File: apps/web/components/CheckoutAddress/CheckoutAddress.vue [hunk 2]
```diff
@@ -37,17 +45,55 @@
...
+  editMode.value = false;
+  open();
+}
+
+function edit() {
```
Comment (ksted) [inline apps/web/components/CheckoutAddress/CheckoutAddress.vue:84]:
```suggestion
const edit = () => {
```

File: apps/web/components/CheckoutAddress/CheckoutAddress.vue [hunk 3]
```diff
@@ -37,17 +45,55 @@
...
+  editMode.value = true;
+  open();
+}
+
+async function saveAddress(address: Address, useAsShippingAddress: boolean = false) {
```
Comment (ksted) [inline apps/web/components/CheckoutAddress/CheckoutAddress.vue:89]:
```suggestion
const saveAddress = async (address: Address, useAsShippingAddress: boolean = false) => {
```

File: apps/web/components/AddressForm/AddressForm.vue [hunk 4]
```diff
@@ -14,53 +14,73 @@
     </label>
     <label class="md:col-span-3">
       <UiFormLabel>{{ $t('form.phoneLabel') }}</UiFormLabel>
-      <SfInput name="phone" type="tel" auto-complete="tel" v-model="defaultValues.phone" required />
+      <SfInput
+        name="phone"
+        type="tel"
+        minlength="7"
+        maxlength="15"
```
Comment (ksted) [inline apps/web/components/AddressForm/AddressForm.vue:21]: **question**: Are we sure we want this? I didn't see this in the requirements and at least `maxlength` feels counterintuitive to me.

File: apps/web/components/CheckoutAddress/CheckoutAddress.vue [hunk 5]
```diff
@@ -2,21 +2,29 @@
   <div data-testid="checkout-address" class="md:px-4 py-6">
     <div class="flex justify-between items-center">
       <h2 class="text-neutral-900 text-lg font-bold mb-4">{{ heading }}</h2>
-      <SfButton v-if="savedAddress" size="sm" variant="tertiary" @click="open">
+      <SfButton v-if="addresses.length > 0" size="sm" variant="tertiary" @click="edit">
```
Comment (ksted) [inline apps/web/components/CheckoutAddress/CheckoutAddress.vue:5]: **suggestion (non-blocking)**: I think it's nice to have a `savedAddress` to clearly express what it means if `length > 0`.

We also check this in lines 10 and 26 in combination with authentication. When dealing with a combined condition, I feel like having a named function improves readability even more.

File: apps/web/components/CheckoutAddress/CheckoutAddress.vue [hunk 6]
```diff
@@ -2,21 +2,29 @@
...
+          {{ `${userAddressGetters.getFirstName(address)} ${userAddressGetters.getLastName(address)}` }}
+        </option>
+      </SfSelect>
     </div>
 
-    <div v-else class="w-full md:max-w-[520px]">
-      <p>{{ description }}</p>
-      <SfButton class="mt-4 w-full md:w-auto" variant="secondary" @click="open">
+    <div v-if="addresses.length === 1 && !isAuth" class="mt-2 md:w-[520px]">
+      <p>{{ `${userAddressGetters.getFirstName(addresses[0])} ${userAddressGetters.getLastName(addresses[0])}` }}</p>
```
Comment (ksted) [inline apps/web/components/CheckoutAddress/CheckoutAddress.vue:19]: **suggestion (non-blocking)**: Extract `addresses[0]` to something like `currentAddress` to improve readability.

**Review by ksted — APPROVED** [review]

File: apps/web/components/AddressForm/AddressForm.vue [hunk 7]
```diff
@@ -72,32 +93,50 @@
   </form>
 </template>
 <script lang="ts" setup>
+import { computed } from 'vue';
```
Comment (maxiroellplenty) [inline apps/web/components/AddressForm/AddressForm.vue:96]: can be removed :)

**Review by maxiroellplenty — APPROVED** [review]

**Review by ksted — APPROVED** [review]

**Pull request merged**